src/integrations/cleanup-integration.php

```diff
@@ -41,10 +42,69 @@ public function cleanup_obsolete_indexables( $object_type, $object_sub_type ) {
 		$number_of_deletions = $this->clean_indexables_with_object_type( $object_type, $object_sub_type, 1000 );
 
 		if ( empty( $number_of_deletions ) ) {
-			$this->unschedule_cron();
+			$this->unschedule_cron( 'wpseo_cleanup_indexables' );
 		}
 	}
 
+	/**
+	 * Cleans orphaned rows from the yoast tables and unregisters the cron if no deletions.
+	 *
+	 * @return void
+	 */
+	public function cleanup_orphaned_indexables() {
+		$deleted_orphans  = $this->cleanup_orphaned_from_table( 'Indexable_Hierarchy', 'indexable_id', 1000 );
+		$deleted_orphans += $this->cleanup_orphaned_from_table( 'SEO_Links', 'indexable_id', 1000 );
+		$deleted_orphans += $this->cleanup_orphaned_from_table( 'SEO_Links', 'target_indexable_id', 1000 );
+
+		$deleted_orphans = \apply_filters( 'wpseo_cleanup_orphaned', $deleted_orphans );
+
+		if ( empty( $deleted_orphans ) ) {
+			$this->unschedule_cron( 'wpseo_cleanup_orphaned_indexables' );
+		}
+	}
+
+	/**
+	 * Cleans orphaned rows from a yoast table.
+	 *
+	 * @param string $table  The table to cleanup.
+	 * @param string $column The table column the cleanup will rely on.
+	 * @param int    $limit  The limit we'll apply to the queries.
+	 *
+	 * @return int The number of deleted rows.
+	 */
+	public function cleanup_orphaned_from_table( $table, $column, $limit = 1000 ) {
+		global $wpdb;
+
+		$table           = Model::get_table_name( $table );
+		$indexable_table = Model::get_table_name( 'Indexable' );
+		$limit           = \apply_filters( 'wpseo_cron_query_limit_size', $limit );
+
+		// Warning: If this query is changed, make sure to update the query in cleanup_orphaned_from_table in Premium as well.
+		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared -- Reason: There is no unescaped user input.
+		$query = $wpdb->prepare(
+			"
+			SELECT h.{$column}
```

Comment:
I'd like a more descriptive alias instead of h / i

Reply:
You're 100% right!